app/frontend/components/pages/receive_ada/addressDetailDialog.js

```diff
@@ -1,96 +1,125 @@
-            {class: 'full-address-row'},
-            h(
-              'span',
-              {
-                class: `full-address ${showVerification ? 'no-select' : 'selectable'}`,
+            'span',
+            {
+              class: 'overlay-close-button',
+              onClick: closeAddressDetail,
+              tabIndex: 3,
+              onKeyDown: (e) => {
+                ;['Enter', ' '].includes(e.key) && e.target.click()
+                if (e.key === 'Tab') {
+                  e.preventDefault()
+                  this.copyBtn.focus()
+                }
               },
-              showDetail.address
-            )
+            },
+            h(CloseIcon)
           ),
-          showVerification
-            ? h(
+          h(
+            'div',
+            undefined,
+            h('b', undefined, 'Address:'),
+            h(
               'div',
-              undefined,
-              h('b', undefined, 'Derivation path:'),
+              {class: 'full-address-row'},
               h(
+                'span',
+                {
+                  class: `full-address ${showVerification ? 'no-select' : 'selectable'}`,
+                },
+                showDetail.address
+              )
+            ),
+            showVerification && showDetail.bip32path
+              ? h(
                 'div',
-                {class: 'full-address-row'},
-                h('span', {class: 'full-address'}, showDetail.bip32path)
-              ),
-              h(
-                'b',
                 undefined,
-                'Verify that the address and derivation path shown on Trezor matches!'
+                h('b', undefined, 'Derivation path:'),
+                h(
+                  'div',
+                  {class: 'full-address-row'},
+                  h('span', {class: 'full-address'}, showDetail.bip32path)
+                ),
+                h(
+                  'b',
+                  undefined,
+                  'Verify that the address and derivation path shown on Trezor matches!'
+                )
               )
-            )
-            : h(
-              'div',
-              undefined,
-              h(
+              : h(
                 'div',
-                {class: 'centered-row'},
-                h(CopyOnClick, {value: showDetail.address}),
+                undefined,
                 h(
-                  Tooltip,
-                  {tooltip: 'Examine via CardanoExplorer.com'},
-                  h(
-                    'a',
-                    {
-                      href: `https://cardanoexplorer.com/address/${showDetail.address}`,
-                      target: '_blank',
-                      class: 'address-link margin-1rem centered-row',
+                  'div',
+                  {class: 'centered-row'},
+                  h(CopyOnClick, {
+                    value: showDetail.address,
+                    tabIndex: 1,
+                    reference: (element) => {
```

Comment:
I would rename this to copyBtnRef

Reply:
I would try assigning this "h(CopyOnClick...)" to a variable, let's say copyOnClick and maybe then you would be able to call focus on it, without this weird passing of reference